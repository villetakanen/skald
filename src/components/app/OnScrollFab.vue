<template>
  <v-btn
    v-scroll="onScroll"
    v-show="scrollFab"
    fab
    fixed
    bottom
    right
    color="primary"
    @click="toTop"
    ><v-icon>mdi-arrow-up-thick</v-icon></v-btn>
</template>

<script lang="ts">
import { defineComponent, ref } from '@vue/composition-api'
export default defineComponent({
  setup () {
    const scrollFab = ref(false)
    const onScroll = (event:{target:HTMLElement}) => {
      if (typeof window === 'undefined') return
      const top = window.pageYOffset || event.target.scrollTop || 0
      scrollFab.value = top > 20
    }
    const toTop = () => window.scroll(0, 0)
    return { scrollFab, onScroll, toTop }
  }
})
</script>
