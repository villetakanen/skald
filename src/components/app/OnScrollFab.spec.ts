import { shallowMount } from '@vue/test-utils'
import OnScrollFab from './OnScrollFab.vue'

describe('Mounted App', () => {
  const wrapper = shallowMount(OnScrollFab)

  it('renders correctly', () => {
    expect(wrapper.element).toMatchSnapshot()
  })
})
