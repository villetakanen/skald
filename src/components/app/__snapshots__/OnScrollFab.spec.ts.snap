// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Mounted App renders correctly 1`] = `
<v-btn-stub
  activeclass=""
  bottom="true"
  color="primary"
  fab="true"
  fixed="true"
  right="true"
  style="display: none;"
  tag="button"
  type="button"
>
  <v-icon-stub>
    mdi-arrow-up-thick
  </v-icon-stub>
</v-btn-stub>
`;
